agent/bench-scripts/pbench-uperf

```diff
@@ -574,11 +578,8 @@ for protocol in `echo $protocols | sed -e s/,/" "/g`; do
 								mkdir -p $benchmark_results_dir
 								server_nr=1
 								# the following loop does all of the pre-benchmark execution work
-								for server in `echo $servers | sed -e s/,/" "/g`; do
-									client=`echo $clients | cut -d, -f$server_nr`
-									if [ -z "$client" ]; then
-										client="127.0.0.1"
-									fi
+								for server in ${servers//,/ }; do
+									client=`echo ${clients} | cut -d, -f$server_nr`
```

Comment:
And, as I think about it, we should be able to get rid of the subcommand altogether and do this directly in the shell:
```
									declare -a client_list=(${clients//,/ })
									client=${client_list[$server_nr]}
```
Note that the array is zero-based, while `cut`'s `-f` is one-based, so if you switch to the code here you'll need to change line 579 to initialize `server_nr` to zero instead of `1`.  And, of course, there's no need to put the assignment of `client_list` inside the loop -- it should be done above in some common location.

In fact, instead of keeping the client list as a comma-separated string and iterating over it using `${clients//,/ }`, we could instead keep it as an array and iterate over it using `${clients[@]}`.

Reply:
Ditto for `vsock_servers` at line 583 and `server_node` at line 603; and also for `client_nodes` at line 620...and many other places...OK, nevermind...it was a nice thought.

Reply:
This is the kind of change that I should make in a separate commit in this PR, kept separate, or in a separate PR altogether, but it would be a good change to make, as I can't stand the gratuitous use of pipes to manipulate strings.

Reply:
Or we just re-write in Python3.

Reply:
A separate, subsequent PR would be excellent.